### DISCUSSION TITLE
Issue with VectorTile generation: Features relocated and shapes incorrect

### DISCUSSION THREAD

**Discussion by miganmht** [NetTopologySuite/NetTopologySuite#756] [Q&A] [answered]
I'm encountering an issue while using the NetTopologySuite.IO.VectorTile package for VectorTile generation in my project. Specifically, I'm observing that some features are being relocated and their shapes are incorrect in the output vector tiles.
I'm using the VectorTileTree to generate vector tiles from GeoJSON data.
Before passing the GeoJSON data to the VectorTileTree, I've verified that the features are correctly positioned and shaped.
However, after generating the vector tiles, I've noticed that some features are relocated to other places and their shapes are incorrect in the output tiles.
i check this by using GeoJsonWriter:  NetTopologySuite.IO.VectorTiles.GeoJson.GeoJsonTileWriter.Write(tree, savedirectory);

my code

    public class Recipe
    {
        public int version { get; set; }
        public List<Layer> layers { get; set; }
    }

    public class Layer
    {
        // Default constructor
        public Layer()
        {
            // Initialize source data if source path is provided
            if (!string.IsNullOrEmpty(_sourcepath))
            {
                sourecedata = ReadSourceDataFromFile(_sourcepath);
            }
        }
        public string name { get; set; }
        private string _sourcepath;
        public string sourcepath
        {
            get { return _sourcepath; }
            set
            {
                _sourcepath = value;
                // sourecedata = ReadSourceDataFromFile(value);
            }
        }
        public FeatureCollection sourecedata { get; set; }
        public int minzoom { get; set; }
        public int maxzoom { get; set; }

        public FeatureCollection ReadSourceDataFromFile(string filePath)
        {
            // Logic to read data from file and deserialize it
            if (File.Exists(filePath))
            {
                string jsonData = File.ReadAllText(filePath);
                return (new NetTopologySuite.IO.GeoJsonReader()).Read<FeatureCollection>(jsonData);
            }
            else
            {
                throw new Exception($"File Dosent Exist {filePath}"); // or handle accordingly if file doesn't exist
            }
        }
    }

        public static void Build_Tile_WithRecepie(in Recipe recipe, string savedirectory)
        {
            var newsavedirecotry = Path.Combine(savedirectory);
            var tilesDir = new DirectoryInfo(newsavedirecotry);

            if (tilesDir.Exists)
            {
                tilesDir.Delete(true);
            }
            Directory.CreateDirectory(newsavedirecotry);
            DateTime start = DateTime.Now;
             var tree = new VectorTileTree();
            foreach (var layer in recipe.layers)
            {             
                ToFeatureZoomAndLayerFunc configFunc = GetFeatureConfigFunc(layer);
                tree.Add(layer.sourecedata, configFunc);
            }
            DateTime treebuild_end = DateTime.Now;
       
            
            var startwrite = DateTime.Now;
            PyramidVectorTileWriter.Write(tree, newsavedirecotry);
            Console.WriteLine($"tree build in {(treebuild_end - start).TotalSeconds} total second");
            
            NetTopologySuite.IO.VectorTiles.GeoJson.GeoJsonTileWriter.Write(tree, savedirectory);

        }

 private static ToFeatureZoomAndLayerFunc GetFeatureConfigFunc(Layer layer)
        {
            IEnumerable<(IFeature feature, int zoom, string layerName)> config(IFeature feature)
            {
                var filtered = new List<(IFeature, int, string)>();
                for (int zoom = layer.minzoom; zoom <= layer.maxzoom; zoom++)
                {
                
                    filtered.Add((feature, zoom, layer.name));
                }

                return filtered;
            }

            return config;
        }
![CorectShape](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/bfd64f72-5fd6-40fe-a1e7-1cf8fed69ec5)
![where4pointmissing](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/f6ad3c1f-a8d7-4488-9575-f064c07412a0)
![CorrectShapeBefor](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/b71bd9eb-97b0-4b50-9f1b-621f885f90d1)
![after](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/fe193400-7352-4afe-b851-8ce15a7818bf)
![4points_relocated](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/e4c2cf45-0575-44f8-831b-7a7e6ea0e34f)
![thispointdosentexistingeojson](https://github.com/NetTopologySuite/NetTopologySuite/assets/72134763/2db559a1-b568-40a8-86fc-690ce566fc08)

**Comment by miganmht** [comment]
I noticed that the geometry of one layer seems to be affecting the geometries of other layers during the vector tile generation process. This inter-layer interference could be causing the misplacement and incorrect shaping of features. Has anyone encountered a similar issue or can provide insights into why geometries from different layers might be interfering with each other?

**Reply by FObermaier** [reply, marked as answer]
This is related to NetTopologySuite/NetTopologySuite.IO.VectorTiles#29

**Reply by miganmht** [reply]
Thank you. The change you made in the other branch worked fine
